src/domain/usecases/user-monitoring/permission-fixer/RunUserPermissionUseCase.ts

```diff
@@ -0,0 +1,454 @@
+                );
+                throw new Error("User: " + user.username + " don't have valid groups");
+            }
+        });
+    }
+
+    async processUserGroups(
+        options: PermissionFixerMetadataConfig,
+        completeTemplateGroups: PermissionFixerTemplateGroupExtended[],
+        allUsersGroupCheck: PermissionFixerUser[]
+    ): Async<PermissionFixerReport> {
+        const {
+            minimalGroupId,
+            permissionFixerConfig: { pushFixedUserGroups },
+        } = options;
+        const response = await this.addLowLevelTemplateGroupToUsersWithoutAny(
+            completeTemplateGroups,
+            allUsersGroupCheck,
+            minimalGroupId,
+            pushFixedUserGroups
+        );
+
+        return response;
+    }
+
+    private async addLowLevelTemplateGroupToUsersWithoutAny(
+        completeTemplateGroups: PermissionFixerTemplateGroupExtended[],
+        allUsersGroupCheck: PermissionFixerUser[],
+        minimalGroupId: NamedRef,
+        pushFixedUserGroups: boolean
+    ): Async<PermissionFixerReport> {
+        const userIdWithoutGroups: NamedRef[] = this.detectUserIdsWithoutGroups(
+            completeTemplateGroups,
+            allUsersGroupCheck,
+            minimalGroupId
+        );
+        return await this.pushUsersWithoutGroupsWithLowLevelGroup(
+            userIdWithoutGroups,
+            minimalGroupId,
+            pushFixedUserGroups
+        );
+    }
+
+    private detectUserIdsWithoutGroups(
+        completeTemplateGroups: PermissionFixerTemplateGroupExtended[],
+        allUsersGroupCheck: PermissionFixerUser[],
+        minimalGroupId: NamedRef
+    ): NamedRef[] {
+        return _.compact(
+            allUsersGroupCheck.map((user): NamedRef | undefined => {
+                const templateGroupMatch = completeTemplateGroups.find(template => {
+                    return user.userGroups.some(
+                        userGroup => userGroup != undefined && template.group.id == userGroup.id
+                    );
+                });
+
+                if (templateGroupMatch == undefined) {
+                    //template not found -> all roles are invalid except the minimal role
+                    log.error(
+                        `Warning: User don't have groups ${user.id} - ${user.name} adding to minimal group  ${minimalGroupId}`
+                    );
+                    const id: NamedRef = { id: user.id, name: user.username };
+                    return id;
+                }
+            })
+        );
+    }
+
+    private async pushUsersWithoutGroupsWithLowLevelGroup(
+        userIdWithoutGroups: NamedRef[],
+        minimalGroupId: NamedRef,
+        pushFixedUserGroups: boolean
+    ): Async<PermissionFixerReport> {
+        if (userIdWithoutGroups != undefined && userIdWithoutGroups.length > 0) {
+            const minimalUserGroup = await this.userGroupRepository.getByIds([minimalGroupId.id]);
```

Comment:
`minimalUserGroup` should be `PermissionFixerUserGroupExtended`, not `PermissionFixerUserGroupExtended[]`.
This will avoid the `[0]!` hacks below.